### PR TITLE
webgui: Introducing CEF and Qt5 implementation of ROOT7 Canvas

### PR DESCRIPTION
**Pull request by linev**
How to compile code see Readme in gui/canvaspainter

Code is not yet used - I will provide another patch after this PR and #717 are merged.

CEF-related makefile should be improved. 
One can use **cefsimple** cmake files as an example

https://bitbucket.org/chromiumembedded/cef/src/8e69e3dcea8b/tests/cefsimple/?at=master

**cefsimple** has some specialization for linux, mac and windows. 
I used only linux till now. One could try mac - for that one should include simple_handler_mac.mm into compilation instead of simple_handler_linux.cxx.

Same is for cef_main.cxx. Formally one need to create separate versions for mac, linux and windows.
You can see example also in **cefsimple**. These are different platform-specific error handlers.

Building of Qt5 part is not called from makefile. I hope, one can add it into makefile when qt5-webengine is discovered. Also rootqt5 executable should be moved into $ROOTSYS/bin directory

### CODE REVIEW

**Comment by linev** [comment]
Current build is based on CEF_PATH shell variable, which is pointed on compiled version of CEF binaries. It should be solved differently.

File: gui/canvaspainter/v7/cef/cef_main.cxx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2013 The Chromium Embedded Framework Authors. All rights
+// reserved. Use of this source code is governed by a BSD-style license that
+// can be found in the LICENSE file.
```
Comment (Axel-Naumann) [inline gui/canvaspainter/v7/cef/cef_main.cxx]: We need that file as gui/canvaspainter/v7/cef/LICENSE

File: gui/canvaspainter/v7/qt5/rootqt5.cpp [hunk 2]
```diff
@@ -0,0 +1,275 @@
+/****************************************************************************
+**
+** Copyright (C) 2016 The Qt Company Ltd.
```
Comment (Axel-Naumann) [inline gui/canvaspainter/v7/qt5/rootqt5.cpp]: This (c) and license looks wrong given the file's content. Can you replace it by a copy&paste of the usual ROOT file header, please?

File: gui/canvaspainter/v7/qt5/rootqt5.pro [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+# RESOURCES += qml.qrc
+
+target.path = .
+# INSTALLS += target
+
+INCLUDEPATH += $$system(root-config --incdir)
+
+LIBS += $$system(root-config --glibs) -lRHTTP
+
+#LIBS += -Wl,-rpath,/home/linev/soft/build6/lib -L/home/linev/soft/build6/lib -lGui -lCore -lRIO -lNet -lHist -lGraf -lGraf3d -lGpad -lTree -lRint -lPostscript -lMatrix -lPhysics -lMathCore -lThread -lMultiProc -pthread -lm -ldl -rdynamic
```
Comment (Axel-Naumann) [inline gui/canvaspainter/v7/qt5/rootqt5.pro]: Could you add a newline?

File: gui/canvaspainter/v7/qt5/rootwebview.h [hunk 4]
```diff
@@ -0,0 +1,36 @@
+#ifndef RootWebView_h
+#define RootWebView_h
+
+#include <QWebEngineView>
+#include <QWebEnginePage>
+
+
+class RootWebPage : public QWebEnginePage {
```
Comment (Axel-Naumann) [inline gui/canvaspainter/v7/qt5/rootwebview.h]: Can you move this class into rootwebview.cpp?

File: gui/canvaspainter/v7/qt5/rootwebview.h [hunk 5]
```diff
@@ -0,0 +1,36 @@
+#ifndef RootWebView_h
```
Comment (Axel-Naumann) [inline gui/canvaspainter/v7/qt5/rootwebview.h]: Can you add the regular ROOT header here and in the cpp file?

File: gui/canvaspainter/v7/cef/cef_main.cxx (hunk 1 above)
Comment (linev) [inline gui/canvaspainter/v7/cef/cef_main.cxx]: I add CEF license file - here we are using cef_simple example practically without any change

File: gui/canvaspainter/v7/qt5/rootqt5.cpp (hunk 2 above)
Comment (linev) [inline gui/canvaspainter/v7/qt5/rootqt5.cpp]: In principle, qt requires to keep there license statement.
From other side, beside few standard construction I fully change code of original qt example.
Therefore I fully remove qt copyright here

File: gui/canvaspainter/v7/qt5/rootqt5.pro (hunk 3 above)
Comment (linev) [inline gui/canvaspainter/v7/qt5/rootqt5.pro]: I change and adjust project file.
To my knowledge, one can provide qt project compilation directly with cmake.

File: gui/canvaspainter/v7/qt5/rootwebview.h (hunk 5 above)
Comment (linev) [inline gui/canvaspainter/v7/qt5/rootwebview.h]: I "ROOT-fy" all headers in qt5

File: gui/canvaspainter/v7/qt5/rootwebview.h (hunk 4 above)
Comment (linev) [inline gui/canvaspainter/v7/qt5/rootwebview.h]: It is not possible - I need to generate Qt dictionary (MOC file) for qt classes to use signal-slot for them. Instead, I extract this class in separate header and source files

**Comment by linev** [comment]
Clang format applied also here.

**Pull request merged**